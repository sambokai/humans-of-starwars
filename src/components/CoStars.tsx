import React from 'react';
import PersonLink from "./PersonLink";

interface OwnProps {
    stars: ReadonlyArray<{
        id: string;
        name: string;
    }>
}

export const CoStars: React.FunctionComponent<OwnProps> = ({stars}) => {

    if (!stars || stars.length === 0) return null;

    return <ul>
        <p>Co-Stars:</p>

        {stars.map(({id, name}) => <li key={id}><PersonLink id={id} name={name}/></li>)}
    </ul>
};
