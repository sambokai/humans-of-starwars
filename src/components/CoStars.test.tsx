import React from 'react';
import {shallow} from "enzyme";

import {CoStars} from "./CoStars";

const mockStars = [
    {id: "1", name: "Luke Skywalker"},
    {id: "2", name: "Jango Fett"},
];

describe('CoStars component', () => {
    describe('when given a list of characters', () => {
        it('should render correctly', () => {
            const wrapper = shallow((
                <CoStars stars={mockStars}/>
            ));

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('when given an empty list', () => {
        it('should render nothing', () => {
            const wrapper = shallow((
                <CoStars stars={[]}/>
            ));

            expect(wrapper.type()).toBeNull();
        });
    });
});
