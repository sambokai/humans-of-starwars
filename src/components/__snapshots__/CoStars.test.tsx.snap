// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoStars component when given a list of characters should render correctly 1`] = `
<ul>
  <p
    title="Characters who appeared in at least 2 consecutive Movies with this character."
  >
    Related Characters:
  </p>
  <li
    key="1"
  >
    <PersonLink
      id="1"
      name="Luke Skywalker"
    />
  </li>
  <li
    key="2"
  >
    <PersonLink
      id="2"
      name="Jango Fett"
    />
  </li>
</ul>
`;
